modules/consentManagement.js

```diff
@@ -81,34 +144,28 @@ function lookupIabConsent(cmpSuccess, cmpError, hookConfig) {
   // check to see if prebid is in a safeframe (with CMP support)
   // else assume prebid may be inside an iframe and use the IAB CMP locator code to see if CMP's located in a higher parent window. this works in cross domain iframes
   // if the CMP is not found, the iframe function will call the cmpError exit callback to abort the rest of the CMP workflow
-  try {
-    cmpFunction = window.__cmp || utils.getWindowTop().__cmp;
-  } catch (e) { }
 
   if (utils.isFn(cmpFunction)) {
-    cmpFunction('getConsentData', null, callbackHandler.consentDataCallback);
-    cmpFunction('getVendorConsents', null, callbackHandler.vendorConsentsCallback);
+    utils.logInfo('Detected CMP API is directly accessible, calling it now...');
+    if (cmpVersion === 1) {
+      cmpFunction('getConsentData', null, v1CallbackHandler.consentDataCallback);
+      cmpFunction('getVendorConsents', null, v1CallbackHandler.vendorConsentsCallback);
+    } else if (cmpVersion === 2) {
+      cmpFunction('addEventListener', cmpVersion, v2CmpResponseCallback);
+    }
   } else if (inASafeFrame() && typeof window.$sf.ext.cmp === 'function') {
-    callCmpWhileInSafeFrame('getConsentData', callbackHandler.consentDataCallback);
-    callCmpWhileInSafeFrame('getVendorConsents', callbackHandler.vendorConsentsCallback);
+    utils.logInfo('Detected Prebid.js is encased in a SafeFrame and CMP is registered, calling it now...');
```

Comment:
Hi @jaiminpanchal27 , Could you please add a comment in code about  "only TCF1 is handled in the case of SafeFrame and in case of TCF2 post-messages will do the job and sf.ext.cmp is not required." Everyone looking at the code may not be aware of the new change :-)

Reply:
Hi @pm-harshad-mane 
Thanks for this feedback.  I did add a comment here to note it was only used for the v1 spec.  However, I also updated the conditional to check if the `cmpVersion` variable was equal to 1 (which should be detected from the `findCMP()` logic); this is to ensure it's only used for the v1 spec.

Reply:
Cool, Thanks @jaiminpanchal27